web-app-router/scenes/Portal/Teams/TeamId/Team/ApiKeys/page/ApiKeyTable/DeleteKeyModal/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { CircleIconContainer } from "@/components/CircleIconContainer";
+import { DecoratedButton } from "@/components/DecoratedButton";
+import { Dialog } from "@/components/Dialog";
+import { DialogOverlay } from "@/components/DialogOverlay";
+import { DialogPanel } from "@/components/DialogPanel";
+import { WarningErrorIcon } from "@/components/Icons/WarningErrorIcon";
+import { TYPOGRAPHY, Typography } from "@/components/Typography";
+import { useDeleteKeyMutation } from "./graphql/client/delete-key.generated";
+import { FetchKeysDocument } from "../../graphql/client/fetch-keys.generated";
+import { toast } from "react-toastify";
+
+type DeleteKeyModalProps = {
+  isOpen: boolean;
+  teamId?: string;
+  keyId?: string;
+  name?: string;
+  setIsOpen: (isOpen: boolean) => void;
+};
+
+export const DeleteKeyModal = (props: DeleteKeyModalProps) => {
+  const { isOpen, teamId, keyId, name, setIsOpen } = props;
+  const [deleteKeyMutation, { loading }] = useDeleteKeyMutation({
+    context: { headers: { team_id: teamId } },
+  });
+
+  const handleDelete = async () => {
+    if (!keyId || loading) {
+      return;
+    }
+    try {
+      const result = await deleteKeyMutation({
+        variables: {
+          id: keyId,
+        },
+        refetchQueries: [
+          {
+            query: FetchKeysDocument,
+            context: { headers: { team_id: teamId } },
```

Comment:
Note this didn't work without the header

Reply:
@andy-t-wang I've just tried it like this `refetchQueries: [FetchKeysDocument]` and it worked for me. Can you, please, doublecheck?

Reply:
@andy-t-wang I've just tried it like this refetchQueries: [FetchKeysDocument] and it worked for me. Can you, please, doublecheck?